types/ember-data/types/registries/model.d.ts

```diff
@@ -1,8 +1,6 @@
-/* eslint @typescript-eslint/no-explicit-any: off */
+/* eslint @typescript-eslint/no-empty-interface: off */
 
 /**
  * Catch-all for ember-data.
  */
-export default interface ModelRegistry {
-  [key: string]: any;
```

Comment:
Removing this line in the addon itself is actually not needed to fix this issue, but doing so reveals the errors that this issue solves in the host app.